source/h2_connection.c

```diff
@@ -1853,6 +1888,13 @@ static void s_move_stream_to_thread(
         goto error;
     }
 
+    if (aws_hash_table_get_entry_count(&connection->thread_data.active_streams_map) == 0) {
+        /* transition from nothing to read -> something to read */
+        uint64_t now_ns = 0;
+        aws_channel_current_clock_time(connection->base.channel_slot->channel, &now_ns);
+        connection->thread_data.incoming_timestamp_ns = now_ns;
+    }
```

Comment:
I said: move this later in the function so we only do it if everything's successful.

then you said we have issues we might have issues removing the stream and doing math but haven't set this

then I said: maybe we use 0 to mean "i got nothing" instead of checking whether the map is empty

Reply:
we still need to check the map is empty or not as the there could be multiple streams created and completed, only when we flip from something to read to nothing to read, we need to do the math, vise verse.

But, we could add a double check using 0.